Fix build error on MySQL version >= 8.0.1

DanielBallaSZTE opened the pull request:
According to [this issue](https://bugs.mysql.com/bug.php?id=85131) in newer versions of MySQL `my_bool` has been removed, and replaced with `bool` type.
On Ubuntu 20.04 OregonCore doesn't compile with the latest packages without this fix.

Can someone check if it brakes anything on previous versions of Ubuntu? I can check Windows in a few hours and report back.

DanielBallaSZTE commented:
~We could possibly also remove remove `my_true` and just use `true`.~
edit: not relevant anymore

DanielBallaSZTE commented:
@yuanf225 Just checkout my Pull Request and try to build the core.

yuanf225 commented:
Travis checks for errors, which will cause compilation failure under Linux. You can fix Travis errors first. I can compile and test under Ubuntu 16.4 and 18.0 when I get home in the evening

DanielBallaSZTE commented:
Yeah Travis just failed on a typecast, I'm gonna fix that fast.

Commit:
Fix build error on MySQL versions 8.01 and above

According to [this issue](https://bugs.mysql.com/bug.php?id=85131) in newer versions of MySQL `my_bool` has been removed, and replaced with `bool` type.
On Ubuntu 20.04 OregonCore doesn't compile with the latest packages without this fix.

talamortis commented:
sorry for the link, @yuanf225 are you able to test any other versions? if all is ok i will merge

DanielBallaSZTE commented:
Looks like my first solution is not going to be backwards compatible. I've updated the PR with a better solution. Fixes the build on 20.04 and doesn't ruin previous versions.

talamortis commented:
so this solution is working?

DanielBallaSZTE commented:
@talamortis Yes, this should be working, however I would be happy if someone else tested on other platforms than Ubuntu 20.04

yuanf225 commented:
Please give me time, I can create ubuntu16/18/20 and centos7, and mysql and MariaDB to verify the compilation

DanielBallaSZTE commented:
That would be amazing, thank you.

yuanf225 commented:
![image](https://user-images.githubusercontent.com/43561197/88752460-8f6af700-d18c-11ea-9ddf-9b70b310385c.png)
I don't have a ready-made system environment to test. Please give me time to deploy them. I'm too busy. Maybe we can give the result tomorrow

DanielBallaSZTE commented:
It's fine, take your time

yuanf225 commented:
Install the project...
-- Install configuration: "Release"
CMake Error at src/server/game/cmake_install.cmake:41 (file):
  file INSTALL cannot find "/root/OregonCore/src/server/game/extensions".
Call Stack (most recent call first):
  src/server/cmake_install.cmake:43 (include)
  src/cmake_install.cmake:45 (include)
  cmake_install.cmake:43 (include)


Makefile:117: recipe for target 'install' failed
make: *** [install] Error 1

again, an error is reported when make install is executed

DanielBallaSZTE commented:
I don't think this issue is related to my PR.

yuanf225 commented:
> 我认为这个问题与我的公关无关。

yeah

yuanf225 commented:
![image](https://user-images.githubusercontent.com/43561197/89015777-33e66800-d34a-11ea-8355-d2b28de7f6ed.png)
![image](https://user-images.githubusercontent.com/43561197/89015764-2d57f080-d34a-11ea-8140-80f14e6a145f.png)
But it failed to compile in Ubuntu20.4 + mysql8.0.21

coolzoom commented:
@yuanf225 Might be your compiler issue. This pr looking good. use gcc -v and g++ -v to check your compiler version. It should compile for gcc. I’m not sure other version

yuanf225 commented:
The tests found that it compiled correctly with GCC, but there were errors with CLang

yuanf225 commented:
![image](https://user-images.githubusercontent.com/43561197/89055127-7f216a80-d38c-11ea-89a2-11719888616b.png)
![image](https://user-images.githubusercontent.com/43561197/89055167-8cd6f000-d38c-11ea-89d5-72992ef2be3c.png)

yuanf225 commented:
System environment: Ubuntu20.4+mysql8.0.21

DanielBallaSZTE commented:
I'm going to try compiling weith Clang, however the compile issues you've attached don't seem to be related to this change.

DanielBallaSZTE commented:
Just compiled with Clang successfully, there seems to be no issue.

talamortis commented:
is this ok to merge and been tested?

yuanf225 commented:
yes
